src/App.js

```diff
@@ -1,7 +1,53 @@
-import React from 'react';
+import React, { Component } from 'react';
+
 import Page from './Page';
 
-const App = (props) => {
-    return <Page data={props.data} />;
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+
+import DotCMSApi from '../src/libs/dotcms.api';
+
+class PageFetchWrapper extends Component {
+    constructor() {
+        super();
+        this.state = {
+            page: {},
+            pathname: null
+        };
+    }
+
+    async setPage(pathname) {
+        const data = await DotCMSApi.getPage({
+            pathname: pathname
+        });
+
+        this.setState({
+            page: data,
+            pathname: pathname
+        });
+    }
+
+    async shouldComponentUpdate(nextProps, prevState) {
+        if (nextProps.location.pathname !== prevState.pathname) {
+            this.setPage(nextProps.location.pathname);
+        }
+
+        return true;
+    }
+
+    async componentDidMount() {
+        this.setPage(this.props.location.pathname);
+    }
+
+    render() {
+        return <Page data={this.state.page.layout} />;
+    }
+}
+
+const App = data => {
+    return (
+        <Router>
```

Comment:
what is the difference between Router and StaticRouter?

https://github.com/dotCMS/dotcms-spa-edit-mode/pull/6/files#diff-ea75b530d74f94670cea053feb8b961bR70

Reply:
Router for client and StaticRouter for server.